src/HexagonAlignLoads.cpp

```diff
@@ -0,0 +1,242 @@
+                        expr = concat_and_shuffle(load_low, load_high, lanes_off, lanes);
+                        debug(4) <<  "... " << expr << "\n";
+                        return;
+                    }
+                } else if (ramp && stride && stride->value == 2) {
+                    // We'll try to break this into two dense loads followed by a shuffle.
+                    Expr base_a = ramp->base, base_b = ramp->base + ramp->lanes;
+                    int b_shift = 0;
+                    int lanes = ramp->lanes;
+
+                    if (op->param.defined()) {
+                        // We need to be able to figure out if buffer_base + base_a is aligned. If not,
+                        // we may have to shift base_b left by one so as to not read beyond the end
+                        // of an external buffer.
+                        AlignCheck ac = get_alignment_check(ramp, op->param.host_alignment(), nullptr);
+
+                        if (ac == AlignCheck::Unaligned) {
+                            debug(4) << "HexagonAlignLoads: base_a is unaligned: shifting base_b\n";
+                            debug(4) << "HexagonAlignLoads: Type: " << op->type << "\n";
+                            debug(4) << "HexagonAlignLoads: Index: " << index << "\n";
+                            base_b -= 1;
+                            b_shift = 1;
+                        }
+                    }
+
+                    Expr ramp_a = Ramp::make(base_a, 1, lanes);
+                    Expr ramp_b = Ramp::make(base_b, 1, lanes);
+                    Expr vec_a = mutate(Load::make(op->type, op->name, ramp_a, op->image, op->param));
+                    Expr vec_b = mutate(Load::make(op->type, op->name, ramp_b, op->image, op->param));
+
+                    std::vector<Expr> indices;
+                    for (int i = 0; i < lanes/2; ++i) {
+                        indices.push_back(i*2);
+                    }
+                    for (int i = lanes/2; i < lanes; ++i) {
+                        indices.push_back(i*2 + b_shift);
+                    }
+
+                    debug(4) << "HexagonAlignLoads: Unaligned Load: Converting " << (Expr) op << " into ...\n";
+
+                    expr = concat_and_shuffle(vec_a, vec_b, indices);
+
+                    debug(4) <<  "... " << expr << "\n";
+                    return;
+                }else {
+                    expr = op;
+                    return;
+                }
+            }
+        } else {
+            expr = op;
+            return;
+        }
+    }
+
+    template<typename NodeType, typename LetType>
+    void visit_let(NodeType &result, const LetType *op) {
+        Expr value;
+        if (op->value.type() == Int(32)) {
+            alignment_info.push(op->name, modulus_remainder(op->value, alignment_info));
+            value = op->value;
+        } else {
+            const Load *ld = op->value.template as<Load>();
+            if (ld) value = mutate(op->value);
```

Comment:
I think this would be a little less surprising to read if it were

    value = ld ? mutate(op->value) : op->value;

Also, it's slightly suspicious to just be mutating loads only. What if the Let were a simple expression of a load?